### PR TITLE
Add new boards for Keychron Q1

### PR DESCRIPTION
**Pull request by lalalademaxiya1**
Added Q1 Version 2 board and made some changes to the original Q1 board (often known as Q1 Version 1).
Thanks!

## Description

- Normally,  Q1 Version 2 is based on STM32L432KBU6 while Q1 Version 1 ATmega32u4. There are still some Q1 Version 2 boards used ATmega32u4. So

1. We moved their unique configuration from the top config.h into the lower config.h

- The matrix size and the led light controller of both versions of Q1 are also different. So

1. We added the mcu name as the suffix to distinguish one from another

- Enable rgb matrix keypress effect for all keyboards, So

1. We added the RGB_MATRIX_KEYPRESSES and RGB_MATRIX_FRAMEBUFFER_EFFECTS to the top config.h

- Fixed the reverse order of the insert and delete key in ansi_encoder_atmega32u4.

- And small format changes

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by jrgiacone** [comment]
can we get this pushed to master?

**Comment by jrgiacone** [comment]
@lalalademaxiya1 I tested out the new firmware for the q1v2, however I am not able to change the knob within VIA. Is there something I need to edit in the json to be able to configure it?

**Comment by adophoxia** [comment]
@jrgiacone, you're probably using a VIA Definitions JSON that's not made for Encoder Mapping support. You'll probably have better luck looking at the docs `[here](https://www.caniusevia.com/docs/layouts#rotary-encoders)` to adapt the JSON to it.

**Comment by K900** [comment]
Here's the Via config I'm using: https://gist.github.com/K900/f24eac85024ac9badfa573e945f12f7f

Note that you need to enable all the lighting effects as I've done [here](https://github.com/K900/qmk_firmware/commit/6d3a544839671501f16436a7b647628c948ac2d6) for the default Via effect names to line up.

**Comment by jrgiacone** [comment]
> @jrgiacone, you're probably using a VIA Definitions JSON that's not made for Encoder Mapping support. You'll probably have better luck looking at the docs [`here`](https://www.caniusevia.com/docs/layouts#rotary-encoders) to adapt the JSON to it.

I changed 0,15 to be 0,15\ne0 and it works as intended now! The only thing that is not present in the firmware here is the reset option from the stock firmware.

File: keyboards/keychron/q1/ansi_stm32l432/rules.mk [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# MCU name
+MCU = STM32L432
+
+# Bootloader selection
+BOOTLOADER = stm32-dfu
+
```
Comment (fauxpark) [inline keyboards/keychron/q1/ansi_stm32l432/rules.mk:6]:
```suggestion
```

File: keyboards/keychron/q1/ansi_stm32l432/info.json [hunk 2]
```diff
@@ -0,0 +1,104 @@
+{
+    "keyboard_name": "Keychron Q1",
+    "manufacturer": "Keychron",
+    "url": "https://github.com/Keychron",
+    "maintainer": "lalalademaxiya1",
+    "usb": {
+        "vid": "0x3434",
+        "pid": "0x0106",
+        "device_version": "2.0.0"
+    },
```
Comment (fauxpark) [inline keyboards/keychron/q1/ansi_stm32l432/info.json:10]:
```suggestion
    },
    "processor": "STM32L432",
    "bootloader": "stm32-dfu",
```

File: keyboards/keychron/q1/ansi_stm32l432_encoder/info.json [hunk 3]
```diff
@@ -0,0 +1,104 @@
+{
+    "keyboard_name": "Keychron Q1",
+    "manufacturer": "Keychron",
+    "url": "https://github.com/Keychron",
+    "maintainer": "lalalademaxiya1",
+    "usb": {
+        "vid": "0x3434",
+        "pid": "0x0107",
+        "device_version": "2.0.0"
+    },
```
Comment (fauxpark) [inline keyboards/keychron/q1/ansi_stm32l432_encoder/info.json:10]:
```suggestion
    },
    "processor": "STM32L432",
    "bootloader": "stm32-dfu",
```

File: keyboards/keychron/q1/ansi_stm32l432_encoder/rules.mk [hunk 4]
```diff
@@ -0,0 +1,32 @@
+# MCU name
+MCU = STM32L432
+
+# Bootloader selection
+BOOTLOADER = stm32-dfu
+
```
Comment (fauxpark) [inline keyboards/keychron/q1/ansi_stm32l432_encoder/rules.mk:6]:
```suggestion
```

File: keyboards/keychron/q1/iso_atmega32u4/config.h [hunk 5]
```diff
@@ -16,11 +16,23 @@
 
 #pragma once
 
+/* key matrix size */
+#define MATRIX_ROWS 6
+#define MATRIX_COLS 15
+
```
Comment (fauxpark) [inline keyboards/keychron/q1/iso_atmega32u4/config.h:22]:
```suggestion
```

File: keyboards/keychron/q1/iso_atmega32u4_encoder/config.h [hunk 6]
```diff
@@ -16,6 +16,10 @@
 
 #pragma once
 
+/* key matrix size */
+#define MATRIX_ROWS 6
+#define MATRIX_COLS 15
+
```
Comment (fauxpark) [inline keyboards/keychron/q1/iso_atmega32u4_encoder/config.h:22]:
```suggestion
```

File: keyboards/keychron/q1/iso_stm32l432/rules.mk [hunk 7]
```diff
@@ -0,0 +1,31 @@
+# MCU name
+MCU = STM32L432
+
+# Bootloader selection
+BOOTLOADER = stm32-dfu
+
```
Comment (fauxpark) [inline keyboards/keychron/q1/iso_stm32l432/rules.mk:6]:
```suggestion
```

File: keyboards/keychron/q1/iso_stm32l432/info.json [hunk 8]
```diff
@@ -0,0 +1,105 @@
+{
+    "keyboard_name": "Keychron Q1",
+    "manufacturer": "Keychron",
+    "url": "https://github.com/Keychron",
+    "maintainer": "lalalademaxiya1",
+    "usb": {
+        "vid": "0x3434",
+        "pid": "0x0108",
+        "device_version": "2.0.0"
+    },
```
Comment (fauxpark) [inline keyboards/keychron/q1/iso_stm32l432/info.json:10]:
```suggestion
    },
    "processor": "STM32L432",
    "bootloader": "stm32-dfu",
```

File: keyboards/keychron/q1/iso_stm32l432_encoder/config.h [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+/* key matrix size */
+#define MATRIX_ROWS 6
+#define MATRIX_COLS 16
+
```
Comment (fauxpark) [inline keyboards/keychron/q1/iso_stm32l432_encoder/config.h:22]:
```suggestion
```

File: keyboards/keychron/q1/iso_stm32l432_encoder/info.json [hunk 10]
```diff
@@ -0,0 +1,105 @@
+{
+    "keyboard_name": "Keychron Q1",
+    "manufacturer": "Keychron",
+    "url": "https://github.com/Keychron",
+    "maintainer": "lalalademaxiya1",
+    "usb": {
+        "vid": "0x3434",
+        "pid": "0x0109",
+        "device_version": "2.0.0"
+    },
```
Comment (fauxpark) [inline keyboards/keychron/q1/iso_stm32l432_encoder/info.json:10]:
```suggestion
    },
    "processor": "STM32L432",
    "bootloader": "stm32-dfu",
```

File: keyboards/keychron/q1/iso_stm32l432_encoder/rules.mk [hunk 11]
```diff
@@ -0,0 +1,32 @@
+# MCU name
+MCU = STM32L432
+
+# Bootloader selection
+BOOTLOADER = stm32-dfu
+
```
Comment (fauxpark) [inline keyboards/keychron/q1/iso_stm32l432_encoder/rules.mk:6]:
```suggestion
```

File: keyboards/keychron/q1/jis_stm32l432/config.h [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+/* key matrix size */
+#define MATRIX_ROWS 6
+#define MATRIX_COLS 16
+
```
Comment (fauxpark) [inline keyboards/keychron/q1/jis_stm32l432/config.h:22]:
```suggestion
```

File: keyboards/keychron/q1/jis_stm32l432/info.json [hunk 13]
```diff
@@ -0,0 +1,108 @@
+{
+    "keyboard_name": "Keychron Q1",
+    "manufacturer": "Keychron",
+    "url": "https://github.com/Keychron",
+    "maintainer": "lalalademaxiya1",
+    "usb": {
+        "vid": "0x3434",
+        "pid": "0x010A",
+        "device_version": "2.0.0"
+    },
```
Comment (fauxpark) [inline keyboards/keychron/q1/jis_stm32l432/info.json:10]:
```suggestion
    },
    "processor": "STM32L432",
    "bootloader": "stm32-dfu",
```

File: keyboards/keychron/q1/jis_stm32l432/rules.mk [hunk 14]
```diff
@@ -0,0 +1,31 @@
+# MCU name
+MCU = STM32L432
+
+# Bootloader selection
+BOOTLOADER = stm32-dfu
+
```
Comment (fauxpark) [inline keyboards/keychron/q1/jis_stm32l432/rules.mk:6]:
```suggestion
```

File: keyboards/keychron/q1/jis_stm32l432_encoder/config.h [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+/* key matrix size */
+#define MATRIX_ROWS 6
+#define MATRIX_COLS 16
+
```
Comment (fauxpark) [inline keyboards/keychron/q1/jis_stm32l432_encoder/config.h:22]:
```suggestion
```

File: keyboards/keychron/q1/jis_stm32l432_encoder/info.json [hunk 16]
```diff
@@ -0,0 +1,108 @@
+{
+    "keyboard_name": "Keychron Q1",
+    "manufacturer": "Keychron",
+    "url": "https://github.com/Keychron",
+    "maintainer": "lalalademaxiya1",
+    "usb": {
+        "vid": "0x3434",
+        "pid": "0x010B",
+        "device_version": "2.0.0"
+    },
```
Comment (fauxpark) [inline keyboards/keychron/q1/jis_stm32l432_encoder/info.json:10]:
```suggestion
    },
    "processor": "STM32L432",
    "bootloader": "stm32-dfu",
```

File: keyboards/keychron/q1/jis_stm32l432_encoder/rules.mk [hunk 17]
```diff
@@ -0,0 +1,33 @@
+# MCU name
+MCU = STM32L432
+
+# Bootloader selection
+BOOTLOADER = stm32-dfu
+
```
Comment (fauxpark) [inline keyboards/keychron/q1/jis_stm32l432_encoder/rules.mk:6]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/keychron/q1/ansi_atmega32u4/config.h [hunk 18]
```diff
@@ -16,11 +16,23 @@
 
 #pragma once
 
+/* key matrix size */
+#define MATRIX_ROWS 6
+#define MATRIX_COLS 15
+
```
[the remark this answers is not in the document]
Comment (lalalademaxiya1) [reply keyboards/keychron/q1/ansi_atmega32u4/config.h]: Thanks~

**Comment by keychron-dev** [comment]
@fauxpark can you have a look at this? thanks

**Comment by keychron-dev** [comment]
@fauxpark hi, please check this one, thanks, a lot of people are waiting for this.

File: keyboards/keychron/q1/ansi_stm32l432_encoder/info.json [hunk 19]
```diff
@@ -0,0 +1,106 @@
...
```
Comment (fauxpark) [inline keyboards/keychron/q1/ansi_stm32l432_encoder/info.json:10]:
```suggestion
    },
    "encoder": {
        "rotary": [
            {"pin_a": "A10", "pin_b": "A8"}
        ]
    },
```

File: keyboards/keychron/q1/ansi_stm32l432_encoder/config.h [hunk 20]
```diff
@@ -0,0 +1,59 @@
...
+#define RGB_MATRIX_LED_COUNT (DRIVER_1_LED_TOTAL + DRIVER_2_LED_TOTAL)
+
+/* EEPROM Driver Configuration */
+#define WEAR_LEVELING_LOGICAL_SIZE 2048
+#define WEAR_LEVELING_BACKING_SIZE (WEAR_LEVELING_LOGICAL_SIZE * 2)
+
+/* Encoder Configuration */
+#define ENCODERS_PAD_A { A10 }
+#define ENCODERS_PAD_B { A8 }
+#define ENCODER_RESOLUTION 4
```
Comment (fauxpark) [inline keyboards/keychron/q1/ansi_stm32l432_encoder/config.h:55]:
```suggestion
```

File: keyboards/keychron/q1/iso_stm32l432_encoder/config.h [hunk 21]
```diff
@@ -0,0 +1,59 @@
...
+/* Scan phase of led driver set as MSKPHASE_9CHANNEL(defined as 0x03 in CKLED2001.h) */
+#define PHASE_CHANNEL MSKPHASE_9CHANNEL
+#define CKLED2001_CURRENT_TUNE \
+    { 0xA6, 0xA6, 0x50, 0xA6, 0xA6, 0x50, 0xA6, 0xA6, 0x50, 0xA6, 0xA6, 0x50 }
+
+/* Encoder Configuration */
+#define ENCODERS_PAD_A { A10 }
+#define ENCODERS_PAD_B { A8 }
+#define ENCODER_RESOLUTION 4
```
Comment (fauxpark) [inline keyboards/keychron/q1/iso_stm32l432_encoder/config.h:51]:
```suggestion
```

File: keyboards/keychron/q1/iso_stm32l432_encoder/info.json [hunk 22]
```diff
@@ -0,0 +1,107 @@
...
```
Comment (fauxpark) [inline keyboards/keychron/q1/iso_stm32l432_encoder/info.json:10]:
```suggestion
    },
    "encoder":  {
        "rotary": [
            {"pin_a": "A10", "pin_b": "A8"}
        ]
    },
```

File: keyboards/keychron/q1/jis_stm32l432_encoder/config.h [hunk 23]
```diff
@@ -0,0 +1,59 @@
...
+/* Scan phase of led driver set as MSKPHASE_9CHANNEL(defined as 0x03 in CKLED2001.h) */
+#define PHASE_CHANNEL MSKPHASE_9CHANNEL
+#define CKLED2001_CURRENT_TUNE \
+    { 0xA0, 0xA0, 0x4D, 0xA0, 0xA0, 0x4D, 0xA0, 0xA0, 0x4D, 0xA0, 0xA0, 0x4D }
+
+/* Encoder Configuration */
+#define ENCODERS_PAD_A { A10 }
+#define ENCODERS_PAD_B { A8 }
+#define ENCODER_RESOLUTION 4
```
Comment (fauxpark) [inline keyboards/keychron/q1/jis_stm32l432_encoder/config.h:51]:
```suggestion
```

File: keyboards/keychron/q1/jis_stm32l432_encoder/info.json [hunk 24]
```diff
@@ -0,0 +1,110 @@
...
```
Comment (fauxpark) [inline keyboards/keychron/q1/jis_stm32l432_encoder/info.json:10]:
```suggestion
    },
    "encoder": {
        "rotary": [
            {"pin_a": "A10", "pin_b": "A8"}
        ]
    },
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by keychron-dev** [comment]
@fauxpark  can you have another look?

**Comment by adophoxia** [comment]
There's still people waiting for this to get merged even for the sake of using VIA due to people getting tired of having to load the JSON every time to use VIA.

**Comment by tzarc** [comment]
Noted.

File: keyboards/keychron/q1/ansi_stm32l432_encoder/keymaps/via/rules.mk [hunk 25]
```diff
@@ -0,0 +1 @@
+ENCODER_MAP_ENABLE = yes
```
Comment (mguentner) [inline keyboards/keychron/q1/ansi_stm32l432_encoder/keymaps/via/rules.mk:1]: `VIA_ENABLE = yes` is missing here

File: keyboards/keychron/q1/matrix.c [hunk 26]
```diff
@@ -0,0 +1,222 @@
...
+#ifdef MATRIX_ROW_PINS
+static pin_t row_pins[MATRIX_ROWS] = MATRIX_ROW_PINS;
+#endif // MATRIX_ROW_PINS
+#ifdef MATRIX_COL_PINS
+static pin_t col_pins[MATRIX_COLS] = MATRIX_COL_PINS;
+#endif // MATRIX_COL_PINS
+
+#define ROWS_PER_HAND MATRIX_ROWS
+
+static inline void setPinOutput_writeLow(pin_t pin) {
```
Comment (mguentner) [inline keyboards/keychron/q1/matrix.c:36]: Suggestion to make function names consistent: convert to underscore
-> `set_pin_output_write_low(pint_t pin)`
(same for the rest of pin functions below)

File: keyboards/keychron/q1/ansi_stm32l432_encoder/keymaps/via/rules.mk (hunk 25 above)
Comment (lalalademaxiya1) [reply keyboards/keychron/q1/ansi_stm32l432_encoder/keymaps/via/rules.mk:1]: Fixed, Thanks!

Comment (lalalademaxiya1) [reply keyboards/keychron/q1/ansi_stm32l432_encoder/keymaps/via/rules.mk:2]: Fixed, Thanks!

File: keyboards/keychron/q1/matrix.c (hunk 26 above)
Comment (lalalademaxiya1) [reply keyboards/keychron/q1/matrix.c:28]: Thanks!

**Comment by keychron-dev** [comment]
@fauxpark hi, will you be able to have some time to check this?

File: keyboards/keychron/q1/ansi_stm32l432/config.h [hunk 27]
```diff
@@ -0,0 +1,53 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+/* key matrix pins */
+#define MATRIX_ROW_PINS { B5, B4, B3, A15, A14, A13 }
+#define MATRIX_COL_PINS { C14, C15, A0, A1, A2, A3, A4, A5, NO_PIN, NO_PIN, NO_PIN, NO_PIN, NO_PIN, NO_PIN, NO_PIN, NO_PIN }
```
Comment (drashna) [inline keyboards/keychron/q1/ansi_stm32l432/config.h:21]: these should be moved to the info.json file. 

Also, the NO_PIN's should be removed

File: keyboards/keychron/q1/ansi_stm32l432_encoder/config.h [hunk 28]
```diff
@@ -0,0 +1,56 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+/* key matrix pins */
+#define MATRIX_ROW_PINS { B5, B4, B3, A15, A14, A13 }
+#define MATRIX_COL_PINS { C14, C15, A0, A1, A2, A3, A4, A5, NO_PIN, NO_PIN, NO_PIN, NO_PIN, NO_PIN, NO_PIN, NO_PIN, NO_PIN }
```
Comment (drashna) [inline keyboards/keychron/q1/ansi_stm32l432_encoder/config.h:21]: same

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/keychron/q1/ansi_stm32l432/config.h (hunk 27 above)
Comment (lalalademaxiya1) [reply keyboards/keychron/q1/ansi_stm32l432/config.h]: Hi, It failed to compile when i removed the NO_PINs from matrix_pins definiton😂.

Comment (lalalademaxiya1) [reply keyboards/keychron/q1/ansi_stm32l432/config.h]: Compile information to remind me that: keychron/q1/ansi_stm32l432: led_config: ansi_stm32l432.c: Number of columns in row 0 (16) does not match matrix (8) and that: keychron/q1/ansi_stm32l432: led_config: ansi_stm32l432.c: Number of columns in row 0 (16) does not match matrix (8)

File: keyboards/keychron/q1/ansi_stm32l432_encoder/info.json [hunk 29]
```diff
@@ -0,0 +1,116 @@
...
+    "matrix_pins": {
+        "cols": ["C14", "C15", "A0", "A1", "A2", "A3", "A4", "A5", "NO_PIN", "NO_PIN", "NO_PIN", "NO_PIN", "NO_PIN", "NO_PIN", "NO_PIN", "NO_PIN"],
```
Comment (vinorodrigues) [inline keyboards/keychron/q1/ansi_stm32l432_encoder/info.json:12]: `null` could (should) be used instead of `"NO_PIN"`.  *(same on the other `json`'s)*

Comment (lalalademaxiya1) [reply keyboards/keychron/q1/ansi_stm32l432_encoder/info.json:12]: All right, Thanks!

Comment (lalalademaxiya1) [reply keyboards/keychron/q1/ansi_stm32l432_encoder/info.json]: Hi, I try to compile it with this way but it was still failed, and displayed that "'null' is not valid under any of the given schemas" in terminal.

File: keyboards/keychron/q1/ansi_stm32l432/config.h (hunk 27 above)
Comment (drashna) [reply keyboards/keychron/q1/ansi_stm32l432/config.h]: There is a higher level that has MATRIX_COLS and MATRIX_ROWS defined?  If so, that would be why

Comment (sigprof) [reply keyboards/keychron/q1/ansi_stm32l432/config.h]: This board uses a custom matrix; pins declared as `NO_PIN` are actually connected to a shift register.  The code is mostly the same as for other similar Keychron boards already in the repo (q10, q12, q3, q5, q6, q65, v1, v10, v3, v5, v6).

**Comment by keychron-dev** [comment]
@drashna  can you have a look to see if it's okay now?

**Comment by adophoxia** [comment]
I still don't think that having each variant of the Q1 be denoted by their MCU would do well when it comes to identifying which model a user has, especially when it comes to troubleshooting since I doubt the average person will open up their board to check what MCU it has so they can flash the correct firmware. A suggestion like delegating each revision into each of their folders like how the [GMMK Pro currently has it](https://github.com/qmk/qmk_firmware/tree/master/keyboards/gmmk/pro/) (Example below):
```
└── Keychron/
    └── q1/
        ├── v1/
        │   ├── ansi
        │   ├── ansi_knob
        │   ├── iso
        │   └── iso_knob
        └── v2/
            ├── ansi
            ├── ansi_knob
            ├── iso
            ├── iso_knob
            ├── jis
            └── jis_knob
``` 
This would look more simplified without having all of that MCU clutter in the way. Some food for thought if this is considered.

**Comment by keychron-dev** [comment]
@drashna can you please help check this one? thanks

**Comment by keychron-dev** [comment]
@drashna will you be able to check it this week?

**Comment by Syphdias** [comment]
Sorry, quick question. I cannot quite tell, if this will also implement support for the Q1 Pro ISO. Is this the case?

**Comment by vinorodrigues** [comment]
No.  Q1-Pro is not in this PR

**Comment by keychron-dev** [comment]
@drashna hi, can you please have a look? thanks

**Comment by keychron-dev** [comment]
can someone check this? it's like been a decade

**Comment by tzarc** [comment]
This constant requesting of reviews doesn't help anyone, and to be honest, makes things worse.

Please refer to the [PR Checklist](https://docs.qmk.fm/#/pr_checklist) for the list of items that need to be handled.

Key points:

* PRs should contain the smallest amount of modifications required for a single change to the codebase
  *  multiple keyboards at the same time is not acceptable
  * **the smaller the PR, the higher likelihood of a quicker review, higher likelihood of quicker merge, and less chance of conflicts**
* `info.json`
  *  _almost all items_, including really obvious things such as:
    * With the move to [data driven](https://docs.qmk.fm/#/data_driven_config) keyboard configuration, we encourage contributors to utilise as many features as possible of the info.json [schema](https://github.com/qmk/qmk_firmware/blob/master/data/schemas/keyboard.jsonschema).
    * If the keyboard only has a single electrical/switch layout:
      * use LAYOUT as your macro name, unless a community layout already exists
* `readme.md`
  * images should be uploaded to an external image hosting service, such as [imgur](https://imgur.com/).

Feel free to go through and address **everything** in the PR checklist and _then_ we'll consider a proper review.

These checklist items have been published for months, and the onus should not be on QMK maintainers to pick out all the issues that have already been published for general consumption. Complaining that nobody reviews when the PR is not in a reviewable state is not QMK's problem.

**Pull request closed without merging**